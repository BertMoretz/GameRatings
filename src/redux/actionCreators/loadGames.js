import axios from "axios"
import * as actionCreators from "./index"


export const loadGamesActionCreator = (values = {}) => (dispatch) => {
    const { query, aliveOnly } = values

    const queryParams = []

    if (query) {
        queryParams.push(`${query}`)
    }

    if (aliveOnly) {
        queryParams.push('status=alive')
    }

    const searchQuery = queryParams.length > 0 ? `?${queryParams.join(' ')}` : ''

    axios({
      url: "https://cors-anywhere.herokuapp.com/http://api-v3.igdb.com/games",
      method: 'POST',
      headers: {
        'Accept': 'application/json',
        'user-key': "861c079a35348acf2360c08a2efc2e90"
      },
      data: `search "${searchQuery}"; fields name,cover.*,aggregated_rating;`
    })
    .then(response => {
      dispatch(actionCreators.gamesListLoaded(response.data))
    })
    .catch((err) => {
      dispatch(actionCreators.gamesListLoadFailed())
    });
}
