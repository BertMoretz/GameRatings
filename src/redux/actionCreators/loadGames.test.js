import axios from 'axios';
import moxios from 'moxios';
import { loadGamesActionCreator } from "./loadGames"


const dispatch = jest.fn()


describe('action loadGamesActionCreator', () => {

    beforeEach(() => {
        moxios.install()

    })

    it('tests moxios', (done) => {

        let onFulfilled = jest.fn()

        moxios.stubRequest('/say/hello', {
            status: 200,
            responseText: 'hello'
        })

        axios.get('/say/hello').then(onFulfilled)

        moxios.wait(function () {
            expect(onFulfilled).toBeCalled()
            console.log('Hmmm')
            done()
        })
    })

    it('works fine', (done) => {

        moxios.stubRequest('https://cors-anywhere.herokuapp.com/http://api-v3.igdb.com/games', {
            status: 200,
            responseText: {
                results: [{ test: 'game' }]
            }
        })

        const dispatcher = loadGamesActionCreator()
        expect(typeof dispatcher).toBe('function')

        dispatcher(dispatch)

        moxios.wait(function () {
            const req = moxios.requests.mostRecent()
            expect(dispatch).toBeCalledTimes(2)
            expect(dispatch.mock.calls[0][0]).toMatchSnapshot()
            done()
        })
    })

    afterEach(() => {
        moxios.uninstall()
    })

})
