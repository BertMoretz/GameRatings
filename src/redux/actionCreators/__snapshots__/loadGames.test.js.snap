// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action loadGamesActionCreator works fine 1`] = `
Object {
  "type": "GAMES_LIST_FETCHING",
}
`;
